src/app/shell/main/main.component.ts

```diff
@@ -34,31 +41,49 @@ export class MainComponent implements OnInit {
   destroy$: Subject<boolean> = new Subject<boolean>();
   @ViewChild('WorkshopsWrap') workshopsWrap: ElementRef;
   public parent: boolean;
+
   getEmptyCards = Util.getEmptyCards;
   widthOfWorkshopCard = Constants.WIDTH_OF_WORKSHOP_CARD;
-  constructor(
-    private store: Store,
-    private actions$: Actions,
-  ) { }
 
+  constructor(private store: Store) { }
+  
+  getTopWorkshops(role: string): void {
+    if(role === Role.parent) {
+      combineLatest([this.city$, this.favoriteWorkshops$])
+      .pipe(
+        filter((result) => {
+          if(!!result[0] && !!result[1]?.length) {
+            return true;
+          }
+          else if(result[1] === null) {
+            return true;
+          }
```

Comment:
I guess you can simplify it to this
filter((result) => (!!result[0] && !!result[1]?.length) || (result[1] === null))

@dmitryy90  verify it please:)

Reply:
not sure what you are trying to achieve with this condition. if workshops === null, the first condition must throw an error.

to avoid using indexes you can use destructing assignment:
const city, workshops;
[city, workshops] = result;